usersession/agent/rest_api.go

```diff
@@ -431,11 +433,43 @@ func postPendingRefreshNotification(c *Command, r *http.Request) Response {
 	return SyncResponse(nil)
 }
 
+func guessAppIcon(si *snap.Info) string {
+	var icon string
+	parser := goconfigparser.New()
+
+	// trivial heuristic, if the app is named like a snap then
+	// it's considered to be the main user facing app and hopefully carries
+	// a nice icon
+	mainApp, ok := si.Apps[si.SnapName()]
+	if ok && !mainApp.IsService() {
+		// got the main app, grab its desktop file
+		if err := parser.ReadFile(mainApp.DesktopFile()); err == nil {
+			icon, _ = parser.Get("Desktop Entry", "Icon")
+		}
+	}
+
+	if icon == "" {
```

Comment:
use reverse logic here to reduce indentation of the code.
```suggestion
	if icon != "" {
	    return icon
        }
```

Reply:
Done.